apps/site/pages/docs/api-extensions/extending-queries-using-fragments.mdx

```diff
@@ -0,0 +1,182 @@
+---
+title: ' Extending queries using fragments'
+sidebar_label: 'API Extensions'
+---
+
+import { Callout, Step, Steps } from 'nextra-theme-docs'
+import { ExtendableQueryTable } from './../../../components/ExtendableQueryTable'
+
+<header>
+
+# Extending queries using fragments
+
+</header>
+
+After [defining the new fields exposed by the FastStore API Extension](/docs/api-extension/extending-api-schema), the next step is to specify where these fields will be utilized within GraphQL fragments.
+
+Fragments in GraphQL are components that enhance query functionality. FastStore associates these fragments with the existing queries used on its pages to incorporate the newly exposed fields.
+
+<Callout type="info" emoji="ℹ️">
+  To access the list of queries and their corresponding fragments, refer to the [extendable queries](/docs/api-extension/extending-queries-using-fragments#extendable-queries) section.
+</Callout>
+
+follow the steps below to add custom fields to an [existing query](/docs/api-extensions#extendable-queries). We will use the `ServerProductPage` query as an example to illustrate how to extend.
+
+---
+## Before you start
+
+<Steps>
+### Avoid over-fetching data on pages
+ 
+Even though you can add information to the FastStore API schema, you must be careful not to over-fetch data on your pages. See the [best practices for fetching data on your storefront](https://v1.faststore.dev/how-to-guides/faststore-api/fetching-api-data#best-practices-for-fetching-data).
+ 
+### Ensure your development server is active
+ 
+Ensure that your development server is up and running while developing to see the changes you make. You can do this by running the command `yarn dev` in your terminal, which will monitor the changes.
```

Comment:
Not necessary as well. If they open the development server before their changes, they'll not be able to see them until they start it again or run faststore generate-graphql.